### DISCUSSION TITLE
LDAP Federation with multiple providers

### DISCUSSION THREAD

**Discussion by roamerz** [keycloak/keycloak#13299] [Q&A]
Hi - I have 3 AD Domains. a.org, b.org and c.org. all three have a user tomsmith. How do I specify which domain to verify against during login? IE tomsmith@b.org

thanks for any help.

**Comment by sonOfRa** [comment]
If you have multiple providers, the order in which they are consulted by keycloak is defined by the "priority" setting in the federation provider settings. The one with the lowest priority is consulted first. I'm not aware of any way to change the order based on the username or other factors.

**Comment by roamerz** [comment]
Thanks sonOfRa. 

Any idea where the code lives that routes the auth requests to the LDAP servers? I will look at that with adding the needed functionality in mind.
